### PR TITLE
Clean-up (lint) of fluent-bit-cloudwatch.go.

### PR DESCRIPTION
**Pull request by davidnewhall**
*Description of changes:*

General cleanup of the main go file and accompanying meta files.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

EDIT:

There's not a lot going on in this PR. I'm OK if you want to just close this. Also willing to make further changes if requested.

### CODE REVIEW

File: fluent-bit-cloudwatch.go [hunk 1]
```diff
@@ -175,10 +182,35 @@ func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int
 	return output.FLB_OK
 }
 
-//export FLBPluginExit
+// FLBPluginExit is exported for fluent-bit.
 func FLBPluginExit() int {
 	return output.FLB_OK
 }
 
+func setupLogger() {
```
Comment (davidnewhall) [inline fluent-bit-cloudwatch.go]: I'm not going to leave this here. I opened https://github.com/aws/amazon-kinesis-firehose-for-fluent-bit/pull/59 for this instead. I'll commit a change to pull this out of here and put this particular setup back how it was. This is where I will test this change again and record the _output_ requested in the linked open PR.

File: fluent-bit-cloudwatch.go [hunk 2]
```diff
@@ -59,79 +58,82 @@ func getPluginInstance(ctx unsafe.Pointer) *cloudwatch.OutputPlugin {
...
-	if val == "true" {
-		return true
-	} else if val == "false" {
-		return false
-	} else {
-		return defaultVal
-	}
-}
-
-//export FLBPluginInit
```
Comment (hossain-rayhan) [inline fluent-bit-cloudwatch.go:109]: Correct me if I am missing something. This `export` comment has syntactical meaning. We cannot change them. This is a `cgo` thing which tells this function will be used from C code I believe.

File: fluent-bit-cloudwatch.go [hunk 3]
```diff
@@ -59,79 +58,82 @@ func getPluginInstance(ctx unsafe.Pointer) *cloudwatch.OutputPlugin {
...
 	if err != nil {
-		logrus.Error(err)
+		logrus.Errorln("addPluginInstance():", err)
 		return output.FLB_ERROR
 	}
+
 	return output.FLB_OK
 }
 
-//export FLBPluginFlushCtx
```
Comment (hossain-rayhan) [inline fluent-bit-cloudwatch.go:121]: Same here. This `export` comment has syntactical meaning. We cannot change them. This is a `cgo` thing which tells this function will be used from C code I believe.

File: fluent-bit-cloudwatch.go [hunk 4]
```diff
@@ -175,10 +182,35 @@ func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int
 	return output.FLB_OK
 }
 
-//export FLBPluginExit
```
Comment (hossain-rayhan) [inline fluent-bit-cloudwatch.go]: Same here

**Review by hossain-rayhan — CHANGES_REQUESTED** [review]

File: fluent-bit-cloudwatch.go (hunk 2 above)
Comment (PettitWesley) [reply fluent-bit-cloudwatch.go:109]: This is the reason why we don't run the linter on the fluent-bit-cloudwatch.go file

Comment (davidnewhall) [reply fluent-bit-cloudwatch.go:109]: oooh, well there ya go. I've not used cgo much at all and had no idea. I'll revert these, thanks!

Comment (davidnewhall) [reply fluent-bit-cloudwatch.go:109]: Can you add to them? ie.
```
// FLBPluginInit is exported for fluent-bit.
//export FLBPluginInit
func FLBPluginInit(ctx unsafe.Pointer) int {
```

Comment (hossain-rayhan) [reply fluent-bit-cloudwatch.go:109]: I would prefer to keep them as they are. Being a little bit conservative. Either way it tells this function is being exported.

**Comment by davidnewhall** [comment]
Closing this in favor of #100. If any of these changes seem beneficial, either _steal_ them, or let me know and I'll re-PR them.

**Pull request closed without merging**